todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/effect_handlers.kt

```diff
@@ -0,0 +1,27 @@
+@file:JvmName("StatisticsEffectHandlers")
+
+package com.example.android.architecture.blueprints.todoapp.statistics
+
+import android.content.Context
+import com.example.android.architecture.blueprints.todoapp.data.source.local.TasksLocalDataSource
+import com.example.android.architecture.blueprints.todoapp.util.schedulers.SchedulerProvider
+import com.spotify.mobius.rx2.RxMobius
+import io.reactivex.ObservableTransformer
+
+fun createEffectHandler(context: Context): ObservableTransformer<StatisticsEffect, StatisticsEvent> {
+    val localDataSource = TasksLocalDataSource.getInstance(context, SchedulerProvider.getInstance())
+    return RxMobius.subtypeEffectHandler<StatisticsEffect, StatisticsEvent>()
+            .addTransformer(StatisticsEffect.LoadTasks::class.java, loadTasksHandler(localDataSource))
+            .build()
+}
+
+fun loadTasksHandler(localSource: TasksLocalDataSource): ObservableTransformer<StatisticsEffect.LoadTasks, StatisticsEvent> {
+    return ObservableTransformer {
+        it.flatMap {
+            localSource.tasks.toObservable()
+                    .take(1)
+                    .map { StatisticsEvent.TasksLoaded(it) as StatisticsEvent }
```

Comment:
Is the casting here needed? It shouldn't be a problem since you've already specified the `StatisticsEvent` type in the function definition, no?

Reply:
I had the same issue but don't really understand why Kotlin can't tell that the subclass `TasksLoaded` is of type `StatisticsEvents`